### ISSUE TITLE
Adaptive-Exploration-for-Unsupervised-Person-Re-Identification/reid/nonparametric_classifier.py, as for line 54 loss = F.cross_entropy(inputs, targets), 'targets' seems like target domain real labels?

### ISSUE THREAD

**Issue by wjm-wjm** [dyh127/Adaptive-Exploration-for-Unsupervised-Person-Re-Identification#4]
Hi, there in the Adaptive-Exploration-for-Unsupervised-Person-Re-Identification/reid/nonparametric_classifier.py, as for line 54: 
![2021-03-27 17-18-46屏幕截图](https://user-images.githubusercontent.com/53390698/112716469-99aee200-8f21-11eb-8c2a-7d7330882958.png)
loss = F.cross_entropy(inputs, targets), 'targets' seems like target domain real labels?

**Comment by dyh127** [comment]
Hi, thank you for your question!

The 'targets' here is the label for different samples instead of different identities, that is one image has its unique label. These labels are easy to obtain. The details can be referred to https://github.com/dyh127/Adaptive-Exploration-for-Unsupervised-Person-Re-Identification/blob/master/reid/datasets/data.py#L78 and https://github.com/dyh127/Adaptive-Exploration-for-Unsupervised-Person-Re-Identification/blob/master/reid/datasets/data.py#L59. 

For the explanation of the code which you are showing, at the beginning of the training (epoch < 4), we feed the target images into the network and force the network to learn to distinguish different images. So that, the network can adapt to the target domain. In this fashion, the network may be able to extract similar features for similar images.  Images with the same identities are usually similar. Thus, when epoch >= 4, we start to minimize feature distances between two similar images which often contain the same person. In this manner, after training, our network can not only distinguish different images but also identify different persons.
